python/cog/server/clients.py

```diff
@@ -288,7 +285,10 @@ async def upload_files(
             with obj.open("rb") as f:
                 return await self.upload_file(f, url=url, prediction_id=prediction_id)
         if isinstance(obj, io.IOBase):
-            return await self.upload_file(obj, url=url, prediction_id=prediction_id)
+            try:
+                return await self.upload_file(obj, url=url, prediction_id=prediction_id)
+            finally:
+                obj.close()
```

Comment:
Is there some reason this doesn't use `with` like the clause above it?

Reply:
Oooh is that what `__wrapped__` does? So that would look like this?

```py
with obj:
  return await self.upload_file(obj, url=url, prediction_id=prediction_id)
```

Nice!